src/gap-skipper.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * @file gap-skipper.js
+ */
+import Ranges from './ranges';
+import videojs from 'video.js';
+
+/**
+ * the gap skipper object handles all scenarios
+ * where the player runs into the end of a buffered
+ * region and there is a buffered region ahead.
+ * It then handles the skipping behavior.
+ *
+ * @class GapSkipper
+ */
+export default class GapSkipper {
+  /**
+   * Represents a GapSKipper object.
+   * @constructor
+   * @param {object} options an object that includes the tech and settings
+   */
+  constructor(options) {
+    if (!options.tech.options_.playerId) {
+      return;
+    }
+
+    this.tech_ = options.tech;
+    this.consecutiveUpdates = 0;
+    this.lastRecordedTime = null;
+    this.timer_ = null;
+
+    if (options.debug) {
+      this.logger_ = videojs.log.bind(videojs, 'gap-skipper ->');
+    }
+    this.logger_('initialize');
+
+    let waitingHandler = () => {
+      if (!this.tech_.seeking()) {
+        this.setTimer_();
+      }
+    };
+
+    // The purpose of this function is to emulate the "waiting" event on
+    // browsers that do not emit it when they are waiting for more
+    // data to continue playback
+    let timeupdateHandler = () => {
+      if (this.tech_.paused() || this.tech_.seeking()) {
+        return;
+      }
+
+      let currentTime = this.tech_.currentTime();
+
+      if (this.consecutiveUpdates === 5 &&
+          currentTime === this.lastRecordedTime) {
+        this.consecutiveUpdates++;
+        waitingHandler();
+      } else if (currentTime === this.lastRecordedTime) {
+        this.consecutiveUpdates++;
+      } else {
+        this.consecutiveUpdates = 0;
+        this.lastRecordedTime = currentTime;
+      }
+    };
+
+    // Set of events that reset the gap-skipper logic and clear the timeout
+    let timerCancelEvents = [
+      'seeking',
+      'seeked',
+      'pause',
+      'playing',
+      'error'
+    ];
+
+    let cancelTimerHandler = this.cancelTimer_.bind(this);
```

Comment:
Why are all the other handlers declared inline in the constructor except for this one?